packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -369,6 +405,76 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
     indent.newln();
   }
 
+  void _writeCodecOverflowUtilities(
+    KotlinOptions generatorOptions,
+    Root root,
+    Indent indent,
+    List<EnumeratedType> types, {
+    required String dartPackageName,
+  }) {
+    final NamedType overflowInt = NamedType(
+        name: 'type',
+        type: const TypeDeclaration(baseName: 'Int', isNullable: false));
+    final NamedType overflowObject = NamedType(
+        name: 'wrapped',
+        type: const TypeDeclaration(baseName: 'Object', isNullable: true));
+    final List<NamedType> overflowFields = <NamedType>[
+      overflowInt,
+      overflowObject,
+    ];
+    final Class overflowClass = Class(
+        name:
+            '${generatorOptions.fileSpecificClassNameComponent}$_overflowClassName',
+        fields: overflowFields);
+
+    indent.writeln('@Suppress("ClassName")');
+    _writeDataClassSignature(indent, overflowClass);
+    indent.addScoped(' {', '}', () {
+      writeClassEncode(
+        generatorOptions,
+        root,
+        indent,
+        overflowClass,
+        dartPackageName: dartPackageName,
+      );
+
+      indent.format('''
+companion object {
+  @Suppress("LocalVariableName")
+  fun fromList(${varNamePrefix}list: List<Any?>): Any? {
+    val wrapper = ${generatorOptions.fileSpecificClassNameComponent}$_overflowClassName(
+      type = ${varNamePrefix}list[0] as Int,
+      wrapped = ${varNamePrefix}list[1],
+    );
+    return wrapper.unwrap()
+  }
+}
+''');
+
+      indent.writeScoped('fun unwrap(): Any? {', '}', () {
+        indent.format('''
+if (wrapped == null) {
+  return null
+}
+    ''');
+        indent.writeScoped('when (type) {', '}', () {
+          for (int i = totalCustomCodecKeysAllowed; i < types.length; i++) {
+            indent.writeScoped('${i - totalCustomCodecKeysAllowed} -> {', '}',
```

Comment:
Since all branches here are a single statement, I don't think the `{}` are necessary.